frontend/resourceadm/pages/DeployResourcePage/DeployResourcePage.tsx

```diff
@@ -26,12 +29,13 @@ export const DeployResourcePage = ({ navigateToPageWithError }: Props) => {
   const { selectedContext, resourceId } = useParams();
   const repo = `${selectedContext}-resources`;
 
-  const [isLoading, setIsLoading] = useState(false);
+  const [loadingPublishStatus, setLoadingPublishStatus] = useState(false);
```

Comment:
It's a lot of local states for status checks, loading checks etc. now. Consider starting to use tanstack query for this instead as this is built in when calling endpoints

Reply:
I will look in to that possibility yes, thanks :)